Code/Engine/InputManager.cpp

```diff
@@ -13,13 +13,63 @@ sad::InputManager& sad::InputManager::GetInstance()
     return instance;
 }
 
-/**
- * @brief Set the controller when a new controller is connected;
- * @param e 
-*/
-void sad::InputManager::OnControllerConnected(SDL_ControllerDeviceEvent& e)
+// Keyboard Events
+
+void sad::InputManager::CatchKeyboardEvents(SDL_Event& event)
 {
-    if (SDL_IsGameController(e.which))
+    if (event.type == SDL_KEYDOWN || event.type == SDL_KEYUP)
+    {
+        keyboardState[event.key.keysym.scancode] = (event.type == SDL_KEYDOWN);
+        keyboardUpdateFrames[event.key.keysym.scancode] = UpdateCounter;
+    }
+}
+
+bool sad::InputManager::GetKey(SDL_Scancode key) 
```

Comment:
This is a general comment that I missed on the first review with William's PR, but do you guys think it'd be beneficial to wrap `SDL_Scancode` with our own enum or small helper class that exposes these??

Something like Unity's `KeyCode` class.

```
namespace sad
{
    enum class KeyCode
    {
         W = SDL_SCANCODE_W,
         A = SDL_SCANCODE_A,
         S = SDL_SCANCODE_S,
         D = SDL_SCANCODE_D
    }
}
```
Then we could refactor the method to be simple like...
```
InputManager::GetInstance().GetKey(sad::KeyCode::W);
```

There's two main benefits to this...

1. We wouldn't have to `#include <SDL2/SDL.h>` in every file and/or gameplay script
2. We would have autocompletion for the keys/buttons that are available to the script/client accessing them (wouldn't have to look up SDL docs for scancodes)

Let me know what you guys think?

Reply:
Yea i agree with this, its added in now. It doesn't map every key though, so I guess we can just add more to the enum as necessary.